### PR TITLE
feat: [BB-4655] Add an admin action to bulk approve users

### PR DESCRIPTION
**Pull request by alfredchavez**
This PR adds an action to Django Admin, so an admin user can bulk approve multiple Student Verification Requests

Jira ticket: [BB-4655](https://tasks.opencraft.com/browse/BB-4655)

Testing instructions:
- Setup custom-student-verification app with a koa devstack
- Add some verification requests(in my case I created a new test user, and uploaded an image for each one in http://localhost:18000/custom-student-verification)
-  Go to http://localhost:18000/admin/custom_student_verification/studentverificationrequest/ to approve the verification requests
- Select some of this requests, then select the "Mark selected student verification requests as ACCEPTED"
- You should see a "success" message at the top, that contains the amount of requests updated
- Verify that the selecred verification requests have an "ACCEPTED" status
- Verify is signals have been triggered, I think you can do it by checking if the send email signal works, look at `/edx/src/ace_messages` and check that the latest files correspond to "ACCEPTED" messages of the verification requests updated

### CODE REVIEW

File: custom_student_verification/admin.py [hunk 1]
```diff
@@ -15,6 +18,7 @@ class StudentVerificationRequestAdmin(admin.ModelAdmin):
     list_display = ('name', 'photo_id_tag', 'id_photo', 'status', 'time_created', 'time_modified')
     readonly_fields = ('name', 'user', 'photo_id_tag', 'id_photo', 'time_created', 'time_modified')
     list_filter = ('status',)
+    actions = ['mark_student_verification_requests_as_accepted']
```
Comment (giovannicimolin) [inline custom_student_verification/admin.py]:
```suggestion
    actions = ['bulk_approve']
```

File: custom_student_verification/admin.py [hunk 2]
```diff
@@ -36,5 +40,60 @@ def get_queryset(self, request):
...
+                request,
+                ngettext(
+                    ' %d Student verification request has failed to update',
+                    ' %d Student verification requests have failed to update',
+                    failed_to_update,
+                ) % failed_to_update,
+                messages.ERROR
+            )
+
+    def mark_student_verification_requests_as_accepted(self, request, queryset):
```
Comment (giovannicimolin) [inline custom_student_verification/admin.py]:
```suggestion
    def bulk_approve(self, request, queryset):
```

File: custom_student_verification/admin.py [hunk 3]
```diff
@@ -36,5 +40,60 @@ def get_queryset(self, request):
...
+        accepted_status = 'ACCEPTED'
+        successfully_updated, failed_to_update = self.update_fields_for_queryset(
+            queryset.exclude(status=accepted_status),
+            'status',
+            accepted_status
+        )
+        self.show_results_to_admin_user(request, successfully_updated, failed_to_update)
+
+    mark_student_verification_requests_as_accepted.short_description = \
+        'Mark selected student verification requests as ACCEPTED'
```
Comment (giovannicimolin) [inline custom_student_verification/admin.py]:
```suggestion
    bulk_approve.short_description = 'Mark selected student verification requests as ACCEPTED'
```

File: custom_student_verification/admin.py [hunk 4]
```diff
@@ -36,5 +40,60 @@ def get_queryset(self, request):
...
+        Given a queryset, this method will update and save a field with a new value on all its records or rows.
+        """
+        successfully_updated_counter = 0
+        failed_to_update_counter = 0
+        for record in queryset:
+            try:
+                setattr(record, field, value)
+                record.save()
+                successfully_updated_counter += 1
+            except IntegrityError as e:
```
Comment (giovannicimolin) [inline custom_student_verification/admin.py:54]: Nit: Why catch `IntegrityError` error here? Can you add a comment as to why catch this?

**Review by giovannicimolin — CHANGES_REQUESTED** [review]
@alfredchavez This is working as expected. I left a few comments in the PR, mostly nits. 
 
Can you few unit tests for these methods?

**Comment by giovannicimolin** [comment]
@alfredchavez This is still missing the unit tests I've requested in the last review.
To make sure this is ready to merge, also add a version bump (`1.3.5`).

**Review by giovannicimolin — CHANGES_REQUESTED** [review]
See my comment above: ^

File: custom_student_verification/__init__.py [hunk 5]
```diff
@@ -4,6 +4,6 @@
 
 from __future__ import absolute_import, unicode_literals
 
-__version__ = '1.2.4'
+__version__ = '1.3.5'
```
Comment (Agrendalath) [inline custom_student_verification/__init__.py]: @alfredchavez, shouldn't this be `1.3.0`?
cc: @giovannicimolin

**Comment by Agrendalath** [comment]
@alfredchavez, @giovannicimolin  do we really need [this step](https://github.com/open-craft/custom-student-verification/blob/main/.circleci/config.yml#L9-L11) in the CI? `virtualenv` is already installed in CircleCI Python images and we don't need to update it, since we're already using the latest Docker images of specific Python versions.

If we really want to do this, we could change `apt-get update` to `apt-get update --allow-releaseinfo-change`, but I would recommend simply deleting this step.

File: custom_student_verification/__init__.py (hunk 5 above)
Comment (giovannicimolin) [reply custom_student_verification/__init__.py]: Yup, the new version should be `1.3.0`.

**Comment by giovannicimolin** [comment]
@Agrendalath I agree and I'll go a bit further: we can delete the Python 3.5 tests since I don't think we don't have any Juniper instances using this module.

**Review by giovannicimolin — APPROVED** [review]
:+1:

- [x] I tested this:
1. On the devstack, following the test instructions.
2. I checked that the signals were properly called for all models on save.
- [x] I read through the code
- [ ] ~~I checked for accessibility issues~~ NA
- [ ] Includes documentation: no, missing documentation. :warning: 

@alfredchavez Nice work, I'm conditionally approving this with two minor changes:
1. Fix the version bump (see comments above).
2. Add usage instructions to https://github.com/open-craft/custom-student-verification/blob/main/doc/Usage.md (they can be basic, but they need to be documented).

Once that is done, you can move this to blocker until the CI is fixed.

**Comment by giovannicimolin** [comment]
@Agrendalath @alfredchavez Here's a PR to simplify and fix the CI checks: https://github.com/open-craft/custom-student-verification/pull/11

**Comment by giovannicimolin** [comment]
@alfredchavez Can you fix the PII issue the CI is failing?

**Comment by alfredchavez** [comment]
@giovannicimolin I've added admin.LogEntry to annotation safeList and avoid the error shown by code_annotations pii_check

**Review by giovannicimolin — APPROVED** [review]
:+1:

- [x] I tested this:
1. Followed the testing instructions, and checked that the signals were properly triggered.
- [x] I read through the code
- [ ] I checked for accessibility issues
- [x] Includes documentation

@alfredchavez Nice work!

**Pull request merged**